src/Commands/RunJobs.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+namespace ActiveCollab\JobQueue\Commands;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use ActiveCollab\JobsQueue\Jobs\Job;
+use ActiveCollab\JobsQueue\Queue\QueueInterface;
+/**
+ * @package ActiveCollab\JobQueue\Commands
+ */
+class RunJobs extends Command
+{
+    /**
+     * Configure command
+     */
+    protected function configure()
+    {
+        parent::configure();
+        $this->setName('run_jobs')
+            ->addOption('seconds', 's', InputOption::VALUE_REQUIRED, 'Run jobs for -s seconds before quitting the process', 50)
+            ->addOption('channels', 'k', InputOption::VALUE_REQUIRED, 'Select one or more channels for jobs for process',QueueInterface::MAIN_CHANNEL)
+            ->setDescription('Run jobs that are next in line for up to N seconds');
+    }
+
+    /**
+     * @param  InputInterface  $input
+     * @param  OutputInterface $output
+     * @return int
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $log = $this->log($input);
+
+        // ---------------------------------------------------
+        //  Connect to database
+        // ---------------------------------------------------
+
+        $connection = $this->getDatabaseConnection($input);
+
+        // ---------------------------------------------------
+        //  Set up environment, so jobs can read Active Collab
+        //  data without directly querying the database
+        // ---------------------------------------------------
+
+        Environment::setDatabaseConnection($connection);
+
+        // ---------------------------------------------------
+        //  Prepare dispatcher and success and error logs
+        // ---------------------------------------------------
+
+        $dispatcher = $this->getDispatcher($input);
+
+        $jobs_ran = $jobs_failed = [];
+
+        $dispatcher->getQueue()->onJobFailure(function (Job $job) use (&$jobs_failed) {
+            $job_id = $job->getQueueId();
+
+            if (!in_array($job_id, $jobs_failed)) {
+                $jobs_failed[] = $job_id;
+            }
+        });
+
+        if ($jobs_count = $dispatcher->getQueue()->count()) {
```

Comment:
Remove this `if` block. Consumer should not exit on empty queue. Instead it should sleep for random time period and wait for new jobs